src/sentry/models/authprovider.py

```diff
@@ -48,5 +61,87 @@ def get_provider(self):
 
         return manager.get(self.provider, **self.config)
 
+    def get_scim_token(self):
+        from sentry.models import SentryAppInstallationForProvider
+
+        if self.flags.scim_enabled:
+            return SentryAppInstallationForProvider.get_token(
+                self.organization, f"{self.provider}_scim"
+            )
+        else:
+            logger.warning(
+                "SCIM disabled but tried to access token",
+                extra={"organization_id": self.organization.id},
+            )
+            return None
+
+    def get_scim_url(self):
+        if self.flags.scim_enabled:
+            return f"https://sentry.io/scim/{self.organization.slug}/scim/v2"
+            # TODO: make dynamic once URL routes are added
+        else:
+            return None
+
+    def enable_scim(self, user):
+        from sentry.mediators.sentry_apps import InternalCreator
+        from sentry.models import SentryAppInstallation, SentryAppInstallationForProvider
+
+        if (
+            not self.get_provider().can_use_scim(self.organization, user)
+            or self.flags.scim_enabled is True
+        ):
+            logger.warning(
+                "SCIM already enabled",
+                extra={"organization_id": self.organization.id},
+            )
+            return
+
+        # check if we have a scim app already
+
+        if SentryAppInstallationForProvider.objects.filter(
+            organization=self.organization, provider="okta_scim"
+        ).exists():
+            logger.warning(
+                "SCIM installation already exists",
+                extra={"organization_id": self.organization.id},
+            )
+            return
+
+        data = {
+            "name": "SCIM Internal Integration",
+            "author": "Auto-generated by Sentry",
+            "organization": self.organization,
+            "overview": SCIM_INTERNAL_INTEGRATION_OVERVIEW,
+            "user": user,
+            "scopes": [
+                "member:read",
+                "member:write",
+                "member:admin",
+                "team:write",
+                "org:write",
+                "org:admin",
+            ],
+        }
+        # create the internal integration and link it to the join table
+        sentry_app = InternalCreator.run(**data)
+        sentry_app_installation = SentryAppInstallation.objects.get(sentry_app=sentry_app)
+        SentryAppInstallationForProvider.objects.create(
```

Comment:
Will note that we're explicitly choosing creating a bearer token over using an OAuth pipeline. -- Other integrations such as gitlab are also doing bearer tokens. Wasn't sure what our praxis was but this seemed reasonable. Interested to hear other's thoughts though.

Reply:
Mainly due to simplicity.